SECURITY.md

```diff
@@ -29,3 +35,89 @@ If you are editing this document please feel encouraged to change this format to
 - #993 Add restricted permissions to RocksDB - 1.4.5 - Anonymous security review
 - #740 Add hardening compile flags and `-fPIE` - 1.4.1 - Anonymous security review
 - #300 Add restricted permissions to osqueryd logging - 1.0.4
+
+## Security Requirements Met by Functionality
+
+### Confidentiality
+
+#### No Retrieval of Sensitive User-owned Data
+
+- Certain tables disabled by default, additional tables can be disabled
+
+#### Use of HTTPS (TLS)
+
+- HTTPS (specifically the TLS v1.2 protocol) is used to encrypt all communications
+
+#### No custom cryptography (only well-known libraries)
+
+#### No network connectivity to third parties
+
+#### Does not accept incoming network connections
+
+#### Does not 'shell out' to other binaries for data collection
+
+#### Access control functions
+
+### Integrity
+
+#### Limited attack surface
+
+#### Daemon runs as root
+
+- Its files are owned as root (config, RocksDB) to resist tampering
+- Extensions must also run as root, communicate over IPC restricted to root
+
+#### No self-update feature
+
+#### No methods to modify data on the system
+
+#### No dynamic linking
+
+#### Code-signed releases
+
+### Availability
+
+#### Minimal configuration by default
+
+#### Watchdog
+
+## Security Implemented in Development Lifecycle Processes
+
+- configuration management via GitHub, can track the changes, who made them, and when they were made
+- we use GitHub for managing the source code and issue tracker; it has an authentication and authorization system
+- limited number of individuals with commit access: [Technical Steering Committee](https://github.com/orgs/osquery/teams/technical-steering-committee), whom have control over the GitHub organization, and whom regularly perform code review and merge actions
+- all changes are made in branches, and then merged only after review
+- second-party review required to merge contribution
+- cppcheck
+- clang-tidy
+- clang sanitizers
+- fuzzed by oss-fuzz ([project page](https://github.com/google/oss-fuzz/tree/master/projects/osquery))
+- modern C++ limiting memory-unsafe language use, not C (except where in third-party library dependencies)
+- warnings as errors: see [the cmake configuration for the project](https://github.com/osquery/osquery/blob/master/cmake/flags.cmake), Wall and pedantic among others are set
+- reproducible builds
+- third-party dependencies retrieved using commit hash
+- opting into compile-time security mitigations (stack protectors, full relro, and other compiler-available hardening options): see [the cmake configuration for the project](https://github.com/osquery/osquery/blob/master/cmake/flags.cmake)
+- how we protect code-signing secrets / key material
+
+## Threat Model
+
+- osquery agent must trust its config server
```

Comment:
And must trust those issuing queries.

Many front-ends and configuration management solutions introduce a user-access-control model and separate out actions like running queries and changing configuration. osquery tries to stay mindful of these downstream tools and make sure they can implement their controls securely. However, osquery does not implement an access control model for configuration or querying.